controllers/idler/idler_controller.go

```diff
@@ -117,11 +122,32 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(ctx, idler, r.setStatusFailed, err,
 			"failed to ensure idling '%s'", idler.Name)
 	}
+	aapIdlerFailed := false
+	if aapIdler, err := newAAPIdler(r.AllNamespacesClient, r.DynamicClient, r.DiscoveryClient, r.notify); err != nil {
+		err := r.wrapErrorWithStatusUpdate(ctx, idler, r.setStatusFailed, fmt.Errorf("failed to init aap idler '%s': %w", idler.Name, err), "")
+		logger.Error(err, "init AAP idler failed")
+		aapIdlerFailed = true
+	} else {
+		aapRequeueAfter, err := aapIdler.ensureAnsiblePlatformIdling(ctx, idler)
+		if err != nil {
+			err := r.wrapErrorWithStatusUpdate(ctx, idler, r.setStatusFailed, fmt.Errorf("failed to ensure aap idling '%s': %w", idler.Name, err), "")
+			logger.Error(err, "AAP idler execution failed")
+			aapIdlerFailed = true
+		}
+		if aapRequeueAfter != 0 {
+			requeueAfter = shorterDuration(requeueAfter, aapRequeueAfter)
+		}
+	}
+
 	logger.Info("requeueing for next pod to check", "after_seconds", requeueAfter.Seconds())
 	result := reconcile.Result{
 		Requeue:      true,
 		RequeueAfter: requeueAfter,
 	}
+	if aapIdlerFailed {
+		// Do not return any errors if the AAP idler failed. We just log the error (above), update the Idler CR status and move on.
```

Comment:
maybe I've missed this but I don't see where the update of the idler CR status happens for aap failures.

Reply:
here https://github.com/codeready-toolchain/member-operator/blob/f169057f4bc8373535fcf67c7403ef25295a0754/controllers/idler/idler_controller.go#L127
and here https://github.com/codeready-toolchain/member-operator/blob/f169057f4bc8373535fcf67c7403ef25295a0754/controllers/idler/idler_controller.go#L133